### DISCUSSION TITLE
Public beta - v2.3.000-rc1

### DISCUSSION THREAD

**Discussion by Medard22** [Medard22/Dolphin-MMJR2-VBI#353] [Announcements]
Changelog from v2.2.110

- Many code refactors including a massive overhaul of graphic pipeline
- Graphic pipeline code revision - removed some old and obsolete hacks
- Updated some external components - SDL, Cubeb, mGBA etc.
- Updated NDK to v25.2.9519653 and some Android libraries
- Minor updates to JIT
- Minor updates to Vulkan backend
- Minor optimization to Vertex Loader
- Updated GameINI database
- Updated out of sync code across the project

This codebase is roughly equivalent to the Official Dolphin dev build 5.0-19479 (Jun 2, 2023).

**General information**
- Hi everyone, It’s been a while since I last worked on the MMJR2 emulator. Unfortunately, work and real-life commitments took a toll on the time I could invest in maintaining MMJR2. I also encountered a significant performance regression that impeded my progress. There are many possible causes for this issue: around June 2023, the Dolphin team added a new asset loader system for graphic mods and custom textures, changed how Dolphin handles threading, and made some updates to the graphics pipeline.
- Since I don’t have as much time anymore, I decided to roll back from Dolphin 5.0-19582 to 5.0-19479 and release a beta version. I made good progress over the months, and it would be a shame not to share it. I did some preliminary testing of this build, and it seems the regressions are not present. However, I didn’t have time to test it thoroughly, hence the beta release.
- I am also waiting for an RP4 Pro and am giving my RP3+ to a friend. While this doesn’t necessarily mean the end of this project, I currently lack a sufficiently low-power device to test new builds on.
- I’ve learned a lot about Git, GitHub, Android development, and Dolphin while working on this project. If you want to help or continue working on MMJR2, let me know. I’d be happy to assist.

**Caution**
- Save States from v2.2.110 are **NOT** compatible with this release. Make sure you have an in-game save before updating.

**Hints**
- For optimal performance on RP3/RP3+ use Performance profile.
- To eliminate shader compilation stuttering enable "Compile Shaders Before Starting". The compiled shaders are usually version/backend specific. Once done you won´t have to compile them again unless you switch to other backend, update the emulator or mess with some settings.

**Known Issues**
- None so far.

<hr /><em>This discussion was created from the release <a href='https://github.com/Medard22/Dolphin-MMJR2-VBI/releases/tag/'>Public beta - v2.3.000-rc1 </a>.</em>

**Comment by divya06-guy** [comment]
hey medard22, i downloaded this version and tried running Need For Speed: Carbon Wii Edition and somehow the cutscenes don't seem to run at all, compared to the previous version of 2.2.110. they just freeze in one frame and the audio just keeps running until the cutscene is over. the gameplay however seems to be running just fine. can you please help me out? im using the same settings on 2.3.000 that i was using on 2.2.110.

**Reply by Medard22** [reply]
There were many changes to the code between the last stable build and this beta. It's hard to tell which one broke it. NFS Carbon uses the infamous EA VP6 codec, I would have to test this game myself.
Try to disable speed hacks and dual core emulation first. I really don't know when I will be able to test that game.

**Reply by Medard22** [reply]
But you could help me out. First I need the ID code of the game to identify your version of NFS : Carbon. Try to disable all speed hacks and dual core and see if it changes anything. If the problem persists you could help me find the issue.

You could install development build of Dolphin Official 5.0-19479. That is roughly the same codebase as my MMJR2 build and test game in it. If it works there it means there is a bug in mmjr2 code and I would have to dig in and compare code between my build and official build to find the cause.

If the problem persists in Dolphin official as well then it's a bug that happened during development and was most likely fixed. You can then uninstall the dev build and download and install an older version - I usually skip back by roughly 10 dev builds. Try the game, rinse and repeat until you find the last working version of Dolphin Official. Once you do, you have the code change that broke the game and we can work with that by identifying changed files and maybe produce a fix.

**Reply by divya06-guy** [reply]
my game ID is RNSE69. and i tried disabling dual core emulation and the emulated clock speed and i got different results. 

by disabling dual core emulation, the game freezes entirely alongwith the audio during startup and no cutscene plays or audio. so disabling dual core emulation only made it worse on my device. my device is an Infinix HOT 10S(Helio G85) by the way. 

by changing emulated clock speed, i got multiple results. it turns out that any value above 60% clock speed seems to make the cutscene run smoothly but this results in a lot of frame skipping and freezing of the cutscene and then the cutscene ends. any value below 60% makes the cutscene run choppy but these values result in much lesser frame skipping and freezing. i had set the value as low as 50% and as high as 100%. values below 50% make the cutscene run extremely choppy and the game tends to crash. values between 80% and 100% freeze the cutscene at the very beginning and only the audio plays. 

i am keeping dual core emulation enabled and have set the emulated clock speed at 60% as it gives workable results for the cutscenes.

**Reply by Medard22** [reply]
I see, helio g85 is even slower than the soc in my RP3+ if underclocking emulated CPU helped then it's more matter of a lack of performance rather than a simple bug and it may have been caused by graphic pipeline overhaul. Unfortunately that's something I cannot fix, because many further fixes and optimizations were built on top of that. But it's good you got it sorted out and thank you for reporting it.

**Reply by XGaming190YT** [reply]
Admin can you please update your dolphin engine to the latest dolphin emulator because that dolphin mmjr you released it was an older dolphin version it seems you're using the previously version and that old version it's kind of poor performance and what ever how my cpu are.and you can update them to latest dolphin released 2407 and that latest dolphin it's included some optimization on some device and some games are no longer have a experiencing frame drops and depends on gpu and i hope you can update them

**Reply by Medard22** [reply]
I am not sure I follow as English is clearly not your first language. Anyway, feel free to use Dolphin Official if that works better for you.

**Comment by JustMeDaFaq** [comment]
Hey @Medard22 
Im using a 60fps gecko code for Super Mario Sunshine. It runs awesome ingame but IT seems to break cutscenes. When the 60fps code ist enabled, cutscenes running at under 20 FPS. Is there any way to fix it?

**Reply by Medard22** [reply]
Well it is technically a hack. Does it work in Dolphin Official?

**Comment by Devbeel** [comment]
the classic button does not work on the wii with dbz tenkaichi 3, please do a new update like on the official dolphin

**Reply by JeffreyProMax** [reply]
Good morning ! I experienced the same problem.  To solve it, I simply switched to the gamecube controls after I simply configured my Controller. the result is excellent ☺️!

**Comment by JeffreyProMax** [comment]
> Journal des modifications de la version 2.2.110
> 
> * De nombreuses refactorisations de code, y compris une refonte massive du pipeline graphique
> * Révision du code du pipeline graphique - suppression de certains hacks anciens et obsolètes
> * Mise à jour de certains composants externes - SDL, Cubeb, mGBA etc.
> * Mise à jour du NDK vers la version 25.2.9519653 et de certaines bibliothèques Android
> * Mises à jour mineures du JIT
> * Mises à jour mineures du backend Vulkan
> * Optimisation mineure de Vertex Loader
> * Mise à jour de la base de données GameINI
> * Mise à jour du code désynchronisé dans l'ensemble du projet
> 
> Cette base de code est à peu près équivalente à la version officielle de développement de Dolphin 5.0-19479 (2 juin 2023).
> 
> **informations générales**
> 
> * Bonjour à tous, Cela fait un moment que je n'ai pas travaillé sur l'émulateur MMJR2. Malheureusement, le travail et les engagements de la vie réelle ont eu un impact sur le temps que je pouvais investir dans la maintenance de MMJR2. J'ai également rencontré une baisse significative des performances qui a entravé ma progression. Il existe de nombreuses causes possibles à ce problème : vers juin 2023, l'équipe Dolphin a ajouté un nouveau système de chargement de ressources pour les mods graphiques et les textures personnalisées, a modifié la façon dont Dolphin gère le threading et a apporté quelques mises à jour au pipeline graphique.
> * Comme je n'ai plus beaucoup de temps, j'ai décidé de revenir de Dolphin 5.0-19582 à 5.0-19479 et de sortir une version bêta. J'ai fait de bons progrès au fil des mois, et il serait dommage de ne pas les partager. J'ai fait quelques tests préliminaires de cette version, et il semble que les régressions ne soient pas présentes. Cependant, je n'ai pas eu le temps de la tester en profondeur, d'où la sortie de la version bêta.
> * J'attends également un RP4 Pro et je donne mon RP3+ à un ami. Bien que cela ne signifie pas nécessairement la fin de ce projet, je manque actuellement d'un appareil suffisamment basse consommation pour tester de nouvelles configurations.
> * J'ai beaucoup appris sur Git, GitHub, le développement Android et Dolphin en travaillant sur ce projet. Si vous souhaitez aider ou continuer à travailler sur MMJR2, faites-le moi savoir. Je serais ravi de vous aider.
> 
> **Prudence**
> 
> * Les états de sauvegarde de la version 2.2.110 ne sont **PAS** compatibles avec cette version. Assurez-vous d'avoir une sauvegarde en jeu avant de procéder à la mise à jour.
> 
> **Astuces**
> 
> * Pour des performances optimales sur RP3/RP3+, utilisez le profil Performance.
> * Pour éliminer les problèmes de compilation des shaders, activez « Compiler les shaders avant de commencer ». Les shaders compilés sont généralement spécifiques à la version/au backend. Une fois cela fait, vous n'aurez plus besoin de les compiler à nouveau, à moins de passer à un autre backend, de mettre à jour l'émulateur ou de modifier certains paramètres.
> 
> **Problèmes connus**
> 
> * Aucun jusqu'à présent.
> 
> _Cette discussion a été créée à partir de la version [Public beta - v2.3.000-rc1](https://github.com/Medard22/Dolphin-MMJR2-VBI/releases/tag/) ._

 @Medard22 

So I tested it, and it works correctly:

🎮 **Games tested**: {DBZ TK3 and SSX Tour}
○ Samsung Galaxy A52s: Snapdragon 778G Adreno 642 ○

● It still works after compiling shaders ✅️😊
● I used Vulkan ✅️
● I increased the image quality to 1.5x, even better ✅️
● I used the pre-installed default settings ✅️

❌️ **DRAWBACKS**

1. 🔴 The application is not detected by Samsung's "Game Plugins" software, unlike the normal version of the Dolphin emulator:

■ This software allows measuring the actual temperature, GPU, and FPS during gameplay. ❌️

2. 🔴 To configure a controller, you have to exit the emulated game each time. ❌️

3. 🔴 There is no possibility to create a controller configuration profile for each game. ❌️

4. 🔴 There is no possibility to create specific settings for each game. ❌️
![Screenshot_20240729_061016_Dolphin MMJR2](https://github.com/user-attachments/assets/b6bcbd83-ee94-4b55-aff5-68b8550cc7ab)
![Screenshot_20240729_062459_Dolphin MMJR2](https://github.com/user-attachments/assets/8682f61a-c0d6-4212-bcd7-bfea11bf4410)
![Screenshot_20240729_065731_Game Plugins](https://github.com/user-attachments/assets/b0e86c6e-7165-49e7-a521-c20d35f4b5dc)
![Screenshot_20240729_062459_Dolphin MMJR2](https://github.com/user-attachments/assets/e979ea02-baa7-4013-a17f-903491a45f21)
![Uploading Screenshot_20240729_061016_Dolphin MMJR2.png…]()

**Reply by Medard22** [reply]
It is possible to make a custom configuration. Long press on a game in the main menu to open a pop-up window. Select "Custom game settings" there, and you can create a custom configuration for that particular game. You can also create a specific controller mapping for any game from the same pop-up window by choosing "Wii Input" or "GameCube Input." This version uses an older version of the Android UI as well as older input handling, so it works differently from how Dolphin Official operates now.

**Reply by JeffreyProMax** [reply]
Okay I'll get back to you after testing

**Comment by JeffreyProMax** [comment]
@Medard22 after the test it turns out that the MMJR2 emulator crashes once a game is launched. I think it's due to the compilation of shaders

**Reply by Medard22** [reply]
Don´t use ubershaders, try to delete shader cache, disable shader precompilation and if all fails post a log.

**Comment by AluZando** [comment]
@Medard22 I installed the new version and it's closing every time I click on the games, it just closes completely! Do you have any solution?

**Reply by Medard22** [reply]
There is a dozen posibilities of what could go wrong. First off make a backup of a mmjr2-vbi folder and delete it, then try again. Set shaders to specialized and don´t use ubershaders (it loves to crash on some GPU drivers). Also try to untick shader precompilation option and let me know if it works. 
When you are able to run games again you can copy back saves from the backup. Then you can experiment with settings. If nothing works enable logging, select all items in logging section, try running couple of games and post a log.

**Reply by AluZando** [reply]
Fiz tudo e mesmo assim continua fechando!

**Comment by Mackzy-vince** [comment]
Can you enhance the Nunchuck control settings by adding support for multiple joysticks dedicated to motion control?😁

**Reply by Medard22** [reply]
I am not sure what do you mean by that, can you elaborate?

**Comment by Geraldy07** [comment]
I suggest that you can install GPU drive like the official dolphin

**Reply by Medard22** [reply]
Custom drivers in MMJR2 won´t happen anytime soon. C++ is okay-ish but Android is a hellscape I can´t wrap my head around. I would have to make a UI for it from scratch. Dolphin Official already moved to Kotlin and heavily changed the Android codebase, so I can´t use their implementation. I would need a help of an skilled Android developer. If you know about any send them my way.

**Comment by Udpota** [comment]
Hello @Medard22 
 I was wondering if you can make an apk with package name "com.antutu.ABenchMark". I tried doing it myself with APK editor but it won't let me open the apk .xml and with a bunch of error, thanks in advance. :)

**Reply by Medard22** [reply]
Ok, this is the latest dev build with id changed to com.antutu.ABenchMark. Completely untested.
https://drive.google.com/file/d/18AcN_gViNMlaLvAuN4KdxVK0xwcKq1NE/view?usp=sharing

**Reply by Udpota** [reply]
Thnks, I'll test it now.

**Reply by Udpota** [reply]
After testing, each time I open the apk it closes.

**Reply by Medard22** [reply]
It seems that just switching the ID is not enough to make it work. It probably won't be that easy. I may look into it, but it's not a high priority for me, sorry.

**Reply by Udpota** [reply]
No worries, but still thanks for your time. :)
